### PR TITLE
Report encryption

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Addresses #1454

## PR Checklist
* [x] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > 
image

### CODE REVIEW

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 1]
```diff
@@ -633,12 +633,11 @@ def generate_report():
...
             },
             "recommendations": {"issues": issues, "domain_issues": domain_issues},
-            "meta": {"latest_monkey_modifytime": monkey_latest_modify_time},
+            "meta_info": {"latest_monkey_modifytime": monkey_latest_modify_time},
         }
         ReportExporterManager().export(report)
-        mongo.db.report.drop()
-        mongo.db.report.insert_one(ReportService.encode_dot_char_before_mongo_insert(report))
-
+        report = ReportService.encode_dot_char_before_mongo_insert(report)
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/reporting/report.py:639]: Move to report model

File: monkey/monkey_island/cc/models/utils/report_encryptor.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+from monkey_island.cc.models.utils.field_types.field_type_abc import FieldTypeABC
+from monkey_island.cc.models.utils.field_types.string_list import StringList
+
+
+@dataclass
+class SensitiveField:
+    path: str
+    path_separator = "."
+    field_type: Type[FieldTypeABC]
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/models/utils/report_encryptor.py:14]: Consider renaming to `field_encryptor`

File: monkey/monkey_island/cc/server_utils/encryption/string_encryptor.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from monkey_island.cc.server_utils.encryptor import get_encryptor
+
+
+def encrypt(cleartext: str) -> str:
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/server_utils/encryption/string_encryptor.py:4]: Remove this layer

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/models/utils/report_encryptor.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+sensitive_fields = [SensitiveField(path="overview.config_passwords", field_type=StringList)]
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/models/utils/report_encryptor.py:17]: Another PR will generalize this file and move this line to report model

File: monkey/monkey_island/cc/models/utils/field_encryptors/string_list_encryptor.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from typing import List
+
+from monkey_island.cc.models.utils.field_encryptors.i_field_encryptor import IFieldEncryptor
+from monkey_island.cc.server_utils.encryptor import get_encryptor
+
+
+class StringListEncryptor(IFieldEncryptor):
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/models/utils/field_encryptors/string_list_encryptor.py:7]: Needs tests

Comment (VakarisZ) [reply monkey/monkey_island/cc/models/utils/field_encryptors/string_list_encryptor.py:7]: Added tests

**Review by ilija-lazoroski — APPROVED** [review]
Amazing work!

**Pull request merged**